# Importing libraries

In [1]:
import pandas as pd
import numpy as np

# Importing databases

In [2]:
pd.set_option('display.max_columns', None)

## Users

In [3]:
users = pd.read_csv("data/exportUser-Grid view.csv")
print(users.shape)
users.head()

(2928, 12)


id                          username  \
0  6005952dcecc894bde4dbe4b  2d76471e6f56a63e6f0105dd92db4254   
1  5e47f02c09468c00014ebdd9  2d76471e6f56a63e6f0105dd92db4254   
2  5ed171d3bc261f00011dcb3c  2d76471e6f56a63e6f0105dd92db4254   
3  5ed505fd719a5e000135831c  2d76471e6f56a63e6f0105dd92db4254   
4  5f0e0c946cae5e0001128704  2d76471e6f56a63e6f0105dd92db4254   

                          firstName                          lastName  \
0  d42c0e46682ec1a404afae4d5cc380e1  e8b55f64846685a7d1b1cf2e6ec63ce4   
1  d42c0e46682ec1a404afae4d5cc380e1  e8b55f64846685a7d1b1cf2e6ec63ce4   
2  d42c0e46682ec1a404afae4d5cc380e1  e8b55f64846685a7d1b1cf2e6ec63ce4   
3  d42c0e46682ec1a404afae4d5cc380e1  e8b55f64846685a7d1b1cf2e6ec63ce4   
4  d42c0e46682ec1a404afae4d5cc380e1  e8b55f64846685a7d1b1cf2e6ec63ce4   

   nbPhoneNumbers  nbEmails city  groupChildSize  groupPersoSize  \
0               0         1  NaN               0               0   
1               0         0  NaN               0               0   
2               0         0  NaN               0               0   
3               0         0  NaN               0               0   
4               0         0  NaN               0               0   

       creationDate lastModificationDate  mapAttributs  
0  2020-01-17 12:18     2021-02-15 14:07           NaN  
1  2020-02-15 14:20     2021-02-15 14:07           NaN  
2  2020-05-01 02:00     2021-02-15 14:07           NaN  
3  2020-05-01 02:00     2021-02-15 14:07           NaN  
4  2020-05-01 02:00     2021-02-15 14:07           NaN

In [4]:
users.groupChildSize.value_counts(dropna=False)

0    2354
1     438
2      99
3      26
4       7
5       4
Name: groupChildSize, dtype: int64

In [5]:
users.city.value_counts(dropna=False)

NaN            2924
Puteaux           1
chatenay          1
panam             1
La Saussaye       1
Name: city, dtype: int64

In [6]:
users.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2928 entries, 0 to 2927
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    2928 non-null   object 
 1   username              2928 non-null   object 
 2   firstName             2928 non-null   object 
 3   lastName              2928 non-null   object 
 4   nbPhoneNumbers        2928 non-null   int64  
 5   nbEmails              2928 non-null   int64  
 6   city                  4 non-null      object 
 7   groupChildSize        2928 non-null   int64  
 8   groupPersoSize        2928 non-null   int64  
 9   creationDate          2928 non-null   object 
 10  lastModificationDate  2928 non-null   object 
 11  mapAttributs          0 non-null      float64
dtypes: float64(1), int64(4), object(7)
memory usage: 274.6+ KB


In [7]:
users.isna().sum()

id                         0
username                   0
firstName                  0
lastName                   0
nbPhoneNumbers             0
nbEmails                   0
city                    2924
groupChildSize             0
groupPersoSize             0
creationDate               0
lastModificationDate       0
mapAttributs            2928
dtype: int64

In [8]:
users.city.unique()

array([nan, 'panam', 'La Saussaye', 'chatenay', 'Puteaux'], dtype=object)

In [9]:
users.nbPhoneNumbers.unique()

array([0, 1])

## Class data

In [10]:
class_data = pd.read_csv("data/exportClassStats-Grid 2020-2021.csv")
print(class_data.shape)
class_data.head()

(313, 28)


id           level                 name schoolYear  \
0  5f49130c4ded6f0001f02861            5ème                   S5  2020-2021   
1  5f4917204ded6f0001f030c0  Petite section                   s5  2020-2021   
2  5f4912c94ded6f0001f027f2  Grande section                   T1  2020-2021   
3  5f49139a4ded6f0001f02a08  Petite section  Classe Madame Sarah  2020-2021   
4  5f4906584ded6f0001f02685  Petite section       non scolarisés  2020-2021   

       creationDate  status  nbChild  schoolid secteur_public_prive_libe  \
0  2020-08-28 16:22     NaN      1.0       NaN                    Public   
1  2020-08-28 16:39     NaN      0.0       NaN                    Public   
2  2020-08-28 16:20     NaN      2.0       NaN                    Public   
3  2020-08-28 16:24     NaN      3.0       NaN                    Public   
4  2020-08-28 15:27     NaN      2.0       NaN                    Public   

            appellation_officielle  code_postal_uai code_commune  \
0              Collège Jean Jaurès            39500        39189   
1  Ecole élémentaire Paul Langevin            39500        39189   
2                      A COMPLETER            98812        98802   
3    Ecole primaire de Boulouparis            98812        98802   
4   Ecole maternelle Louis Pasteur            27370        27616   

  libelle_commune localite_acheminement_uai libelle_departement  \
0        Damparis                  DAMPARIS                Jura   
1        Damparis                  DAMPARIS                Jura   
2      Bouloupari               BOULOUPARIS  Nouvelle Calédonie   
3      Bouloupari               BOULOUPARIS  Nouvelle Calédonie   
4     La Saussaye               LA SAUSSAYE                Eure   

                       libelle_region geometry_type  coordinatesLat  \
0             Bourgogne-Franche-Comté         Point       47.068158   
1             Bourgogne-Franche-Comté         Point       47.069952   
2  TOM et Collectivités territoriales           NaN             NaN   
3  TOM et Collectivités territoriales           NaN             NaN   
4                           Normandie         Point       49.260635   

   coordinatesLong  exportHomeworkRequest lastModificationDate  Table 8  \
0         5.414603                    NaN     2021-02-15 14:07      NaN   
1         5.415737                    NaN     2021-02-15 14:07      NaN   
2              NaN                    NaN     2021-02-15 14:07      NaN   
3              NaN                    NaN     2021-02-15 14:07      NaN   
4         0.977308                    NaN     2021-02-15 14:07      NaN   

   exportHomeworkRequest 2  exportHomeworkRequest 3  \
0                      NaN                      NaN   
1                      NaN                      NaN   
2                      NaN                      NaN   
3                      NaN                      NaN   
4                      NaN                      NaN   

                            coordinates  nbArchivedChildren  \
0   47.06815774787053,5.414603395756424                 0.0   
1   47.06995174190594,5.415737072219595                 0.0   
2                                     ,                 0.0   
3                                     ,                 0.0   
4  49.26063467583511,0.9773083272907583                 0.0   

                                            cacheMap  Table 8 (2)  
0  🔵 eyJpIjoiNDcuMDY4MTU3NzQ3ODcwNTMsNS40MTQ2MDMz...          NaN  
1  🔵 eyJpIjoiNDcuMDY5OTUxNzQxOTA1OTQsNS40MTU3Mzcw...          NaN  
2  🔵 eyJpIjoiLCIsIm8iOnsic3RhdHVzIjoiT0siLCJmb3Jt...          NaN  
3  🔵 eyJpIjoiLCIsIm8iOnsic3RhdHVzIjoiT0siLCJmb3Jt...          NaN  
4  🔵 eyJpIjoiNDkuMjYwNjM0Njc1ODM1MTEsMC45NzczMDgz...          NaN

In [11]:
class_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 313 entries, 0 to 312
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   id                         313 non-null    object 
 1   level                      313 non-null    object 
 2   name                       313 non-null    object 
 3   schoolYear                 313 non-null    object 
 4   creationDate               313 non-null    object 
 5   status                     0 non-null      float64
 6   nbChild                    313 non-null    float64
 7   schoolid                   0 non-null      float64
 8   secteur_public_prive_libe  313 non-null    object 
 9   appellation_officielle     313 non-null    object 
 10  code_postal_uai            313 non-null    int64  
 11  code_commune               313 non-null    object 
 12  libelle_commune            313 non-null    object 
 13  localite_acheminement_uai  313 non-null    object 

In [12]:
(class_data.isna().sum()/class_data.shape[0]).round(2)

id                           0.00
level                        0.00
name                         0.00
schoolYear                   0.00
creationDate                 0.00
status                       1.00
nbChild                      0.00
schoolid                     1.00
secteur_public_prive_libe    0.00
appellation_officielle       0.00
code_postal_uai              0.00
code_commune                 0.00
libelle_commune              0.00
localite_acheminement_uai    0.00
libelle_departement          0.00
libelle_region               0.00
geometry_type                0.01
coordinatesLat               0.01
coordinatesLong              0.01
exportHomeworkRequest        1.00
lastModificationDate         0.00
Table 8                      1.00
exportHomeworkRequest 2      1.00
exportHomeworkRequest 3      1.00
coordinates                  0.00
nbArchivedChildren           0.00
cacheMap                     0.00
Table 8 (2)                  1.00
dtype: float64

## Events

In [13]:
events = pd.read_csv("data/exportEvent-Grid view.csv")
print(events.shape)
events.head()

(2426, 22)


id                      label  \
0  5e6093839fed0e000155fef3  Lancement programmé Start   
1  5e61123e9fed0e00015600a6         Dîner dailyartfair   
2  5e75d5179fed0e000191fcec     Un petit télé apéro ?    
3  5ec28f8b7b059100010baa8f        Côte de boeuf party   
4  5e62410b9fed0e000156035a                 Mix Mondes   

                             description                       location  \
0                        Lancement + dej  Willa 35 rue du Sentier Paris   
1                                    NaN                            NaN   
2                                    NaN                       En visio   
3  La côte de bœuf, c'est bon. Miam miam                    La Saussaye   
4               Atelier pour les enfants               La Gaité lyrique   

   nbPeriods firstPeriods-startDate firstPeriods-endDate  \
0          1       2020-03-11 11:00     2020-03-11 13:30   
1          1       2020-03-06 21:51     2020-03-06 23:51   
2          1       2020-03-22 19:00     2020-03-22 00:00   
3          1       2020-06-01 12:00     2020-06-01 15:00   
4          1       2020-03-15 15:30     2020-03-15 16:30   

   firstPeriods-nbParticipants  firstPeriods-status  firstPeriods-allDay  \
0                          NaN                  NaN                  NaN   
1                          NaN                  NaN                  NaN   
2                          NaN                  NaN                  NaN   
3                          NaN                  NaN                  NaN   
4                          NaN                  NaN                  NaN   

                     author      creationDate lastModificationDate visibility  \
0  5e5ccf59026b4800016b27b0  2020-03-05 06:52     2020-03-05 06:52    PRIVATE   
1  5e5e2cb73dc85e0001e90672  2020-03-05 15:52     2020-03-05 15:52    PRIVATE   
2  5e5ccf59026b4800016b27b0  2020-03-21 09:49     2020-03-21 09:49    PRIVATE   
3  5e5ccf59026b4800016b27b0  2020-05-18 15:37     2020-05-22 10:44    PRIVATE   
4  5e617dc09fed0e00015600b2  2020-03-06 13:24     2020-03-06 21:14    PRIVATE   

    status frequencyType  dayOfWeek dayOfMonth frequencyValue  groupPartSize  \
0  CREATED          ONCE  WEDNESDAY         11              1              2   
1  CREATED          ONCE   SATURDAY          7              1              1   
2  CREATED          ONCE     SUNDAY         22              1              2   
3  CREATED          ONCE     MONDAY          1              1              3   
4  CREATED          ONCE     SUNDAY         15              1              2   

   groupOrgSize eventType  
0             1       NaN  
1             1       NaN  
2             1       NaN  
3             1       NaN  
4             2       NaN

In [14]:
events.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2426 entries, 0 to 2425
Data columns (total 22 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           2426 non-null   object 
 1   label                        2412 non-null   object 
 2   description                  862 non-null    object 
 3   location                     1044 non-null   object 
 4   nbPeriods                    2426 non-null   int64  
 5   firstPeriods-startDate       2420 non-null   object 
 6   firstPeriods-endDate         2420 non-null   object 
 7   firstPeriods-nbParticipants  0 non-null      float64
 8   firstPeriods-status          0 non-null      float64
 9   firstPeriods-allDay          0 non-null      float64
 10  author                       2426 non-null   object 
 11  creationDate                 2426 non-null   object 
 12  lastModificationDate         2426 non-null   object 
 13  visibility        

In [15]:
(events.isna().sum()/events.shape[0]).round(2)

id                             0.00
label                          0.01
description                    0.64
location                       0.57
nbPeriods                      0.00
firstPeriods-startDate         0.00
firstPeriods-endDate           0.00
firstPeriods-nbParticipants    1.00
firstPeriods-status            1.00
firstPeriods-allDay            1.00
author                         0.00
creationDate                   0.00
lastModificationDate           0.00
visibility                     0.14
status                         0.00
frequencyType                  0.00
dayOfWeek                      0.00
dayOfMonth                     0.00
frequencyValue                 0.00
groupPartSize                  0.00
groupOrgSize                   0.00
eventType                      0.54
dtype: float64

In [16]:
events["label"].nunique()/events.shape[0]

0.281121187139324

In [17]:
print(events["label"].unique())

['Lancement programmé Start' 'Dîner dailyartfair' 'Un petit télé apéro ? '
 'Côte de boeuf party' 'Mix Mondes' 'test' 'le 1' 'le 2'
 'piscine les vendredi' 'Pot de départ Budget Box Simon' 'soiree apero'
 'Dîner en Amoureux' 'Semaine de travail au vert' 'Test prod'
 '1 er test caro' 'un evenement recurent' 'Les lyonnais à Paris'
 'Truffes, whisky et cigares' 'Test' 'Lancement de la session Bêta test 2'
 'Dîner marie' 'Test envoi mail' "c'est le week-end !" 'Test Alex'
 'Test rendez vous ' 'Surf' 'Test event' 'Point' 'Pique-nique'
 'anniv annelolo' 'Point digital marketing' 'Test date' 'Marché '
 'Trip à Paris' 'Pendaison de crémaillère' 'Bienvenue à CC'
 'BBQ de rentrée !' 'Déjeuner Séverine-Alex' 'Déjeuner Delphine-Alex'
 'Ça progresse ? ' 'Essai' 'Test mail' 'Dîner entre amis ' 'Coucou '
 'Déjeuner à Paris' "Les parisiens à L'isle-Adam" 'Dîner Danette '
 "A l'abri de la chaleur" 'Dej fin de vacances ' 'Test feamzy'
 'Rdv avec moi' 'Anniversaire marlene et anthony' 'feamzy<>Bluelemon'

In [18]:
print(events.frequencyValue.nunique())
print(events.frequencyValue.unique())

2
['1' 'Null']


In [19]:
print(events.nbPeriods.nunique())
print(events.nbPeriods.unique())

8
[ 1  3  2  0  4  5 10  6]


## Documents

In [20]:
documents = pd.read_csv("data/exportDocument-Grid view.csv")
print(documents.shape)
documents.head()

(51, 13)


id               referenceId referenceType userId  \
0  5f5c8d759a49ce0001c51e0c  5f4e61b8c154d5000100b02a         CLASS   Null   
1  5f5c80c6217d3600011d35bc  5f4e61b8c154d5000100b02a         CLASS   Null   
2  5f5c806e217d3600011d35a9  5f4e61b8c154d5000100b02a         CLASS   Null   
3  5f5c809e217d3600011d35ae  5f4e61b8c154d5000100b02a         CLASS   Null   
4  5f5c8056217d3600011d35a5  5f4e61b8c154d5000100b02a         CLASS   Null   

        label      description           type              creationDate  \
0  Grammaire   desc Grammaire        HOMEWORK  2020-09-12T08:57:25.677Z   
1         NaN              NaN  HOMEWORK_LIST  2020-09-12T08:03:18.005Z   
2         NaN              NaN  HOMEWORK_LIST  2020-09-12T08:01:50.911Z   
3         NaN              NaN  HOMEWORK_LIST  2020-09-12T08:02:38.458Z   
4         NaN              NaN  HOMEWORK_LIST  2020-09-12T08:01:26.221Z   

       lastModificationDate                 echueDate  contentType authorName  \
0  2020-09-12T08:57:25.677Z  2020-09-13T23:00:00.000Z          NaN        NaN   
1  2020-09-12T08:03:18.005Z  2020-09-17T23:00:00.000Z          NaN        NaN   
2  2020-09-12T08:01:50.911Z  2020-09-14T23:00:00.000Z          NaN        NaN   
3  2020-09-12T08:02:38.458Z  2020-09-16T23:00:00.000Z          NaN        NaN   
4  2020-09-12T08:01:26.221Z  2020-09-13T23:00:00.000Z          NaN        NaN   

                   modifier  
0  5e5e2cb73dc85e0001e90672  
1  5e5e2cb73dc85e0001e90672  
2  5e5e2cb73dc85e0001e90672  
3  5e5e2cb73dc85e0001e90672  
4  5e5e2cb73dc85e0001e90672

In [21]:
documents.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51 entries, 0 to 50
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    51 non-null     object 
 1   referenceId           51 non-null     object 
 2   referenceType         51 non-null     object 
 3   userId                51 non-null     object 
 4   label                 21 non-null     object 
 5   description           5 non-null      object 
 6   type                  51 non-null     object 
 7   creationDate          51 non-null     object 
 8   lastModificationDate  51 non-null     object 
 9   echueDate             51 non-null     object 
 10  contentType           0 non-null      float64
 11  authorName            6 non-null      object 
 12  modifier              51 non-null     object 
dtypes: float64(1), object(12)
memory usage: 5.3+ KB


In [22]:
(documents.isna().sum()/documents.shape[0]).round(2)

id                      0.00
referenceId             0.00
referenceType           0.00
userId                  0.00
label                   0.59
description             0.90
type                    0.00
creationDate            0.00
lastModificationDate    0.00
echueDate               0.00
contentType             1.00
authorName              0.88
modifier                0.00
dtype: float64

In [23]:
documents

id               referenceId referenceType  \
0   5f5c8d759a49ce0001c51e0c  5f4e61b8c154d5000100b02a         CLASS   
1   5f5c80c6217d3600011d35bc  5f4e61b8c154d5000100b02a         CLASS   
2   5f5c806e217d3600011d35a9  5f4e61b8c154d5000100b02a         CLASS   
3   5f5c809e217d3600011d35ae  5f4e61b8c154d5000100b02a         CLASS   
4   5f5c8056217d3600011d35a5  5f4e61b8c154d5000100b02a         CLASS   
5   5f6a0ab5fb6eb600010f6ccd  5f6a0a44046ca50001ad3afe         CLASS   
6   5f6c9f8d046ca50001ad3e8f  5f4e27ab2447780001556cf0         CLASS   
7   5f7d7b1f08a33300018cce0a  5f54ce6585b2260001876b6f         CLASS   
8   5f7deff008a33300018cd0e4  5f7def419fe5530001fd2151         CLASS   
9   5f64c4ca8001bd00012c95e4  5f4e27ab2447780001556cf0         CLASS   
10  5f68c1a0046ca50001ad371b  5f4e27ab2447780001556cf0         CLASS   
11  5f68c1d9046ca50001ad372b  5f4e27ab2447780001556cf0         CLASS   
12  5f68dbd1fb6eb600010f696e  5f54ce6585b2260001876b6f         CLASS   
13  5f75a10508a33300018cbe2e  5f4e46f02447780001556df3         CLASS   
14  5f75b80a08a33300018cbfb8  5f5d27d2217d3600011d3701         CLASS   
15  5f79b4389fe5530001fd17e6  5f76283708a33300018cc0f4         CLASS   
16  5f85e3a79fe5530001fd2db7  5f54ce6585b2260001876b6f         CLASS   
17  5f85f2909fe5530001fd2e35  5f54ce6585b2260001876b6f         CLASS   
18  5f491db44ded6f0001f039f3  5f4912c94ded6f0001f027f2         CLASS   
19  5f512e3e2e234000015850ec  5f4cfe154ded6f0001f03f62         CLASS   
20  5f621bd3b4d48f0001db5ab9  5f4e27ab2447780001556cf0         CLASS   
21  5f4914ff4ded6f0001f02d1f  5f4914554ded6f0001f02c8c         CLASS   
22  5f4916a44ded6f0001f02fec  5f4914934ded6f0001f02cb4         CLASS   
23  5f4917d24ded6f0001f0323a  5f4917204ded6f0001f030c2         CLASS   
24  5f4919e64ded6f0001f0346b  5f4914934ded6f0001f02cb4         CLASS   
25  5f6219e9b4d48f0001db5a6e  5f4e27ab2447780001556cf0         CLASS   
26  5f49150f4ded6f0001f02d35  5f49130c4ded6f0001f02861         CLASS   
27  5f79785f9fe5530001fd17ae  5f54ce6585b2260001876b6f         CLASS   
28  5f735583fb6eb600013ddc9f  5f4e27ab2447780001556cf0         CLASS   
29  5f4916984ded6f0001f02fea  5f4914934ded6f0001f02cb4         CLASS   
30  5f4917584ded6f0001f03137  5f4912c94ded6f0001f027f2         CLASS   
31  5f5790089ecb02000102675f  5f4906584ded6f0001f02685         CLASS   
32  5f75897308a33300018cbde0  5f4e27ab2447780001556cf0         CLASS   
33  5fa4e6c36cbf5200019247b9  5f4906584ded6f0001f02685         CLASS   
34  5fa4e7386cbf5200019247cc  5f4906584ded6f0001f02685         CLASS   
35  5fa7fd51f528270001d223e0  5fa7fc69f528270001d223cd         CLASS   
36  5fa7fd496cbf52000192517d  5fa7fc69f528270001d223cd         CLASS   
37  5fa98a08f528270001d22602  5f7def419fe5530001fd2151         CLASS   
38  5fa97672f528270001d22579  5f54ce6585b2260001876b6f         CLASS   
39  5fabac1e6cbf5200019254df  5f54ce6585b2260001876b6f         CLASS   
40  5fabac2ff528270001d22731  5f54ce6585b2260001876b6f         CLASS   
41  5fabbfb9f528270001d227ef  5f4e46f02447780001556df3         CLASS   
42  5fad7bcdf528270001d22f5c  5f54ce6585b2260001876b6f         CLASS   
43  5fad6922f528270001d22ea8  5f54ce6585b2260001876b6f         CLASS   
44  5fb2b4ebf528270001d23361  5f54ce6585b2260001876b6f         CLASS   
45  5fcb4afbd4ce9c0001e01573  5f6ef013046ca50001ad41dd         CLASS   
46  5fce5ee8d4ce9c0001e017b9  5f49139a4ded6f0001f02a08         CLASS   
47  5fdb4e073f4f2900015c502c  5f6ef013046ca50001ad41dd         CLASS   
48  601ab8f187a2df7a6734c23f  601ab8bf87a2df7a6734c22f         CLASS   
49  6026b2452be77d0e8a247923  5f76283708a33300018cc0f4         CLASS   
50  6058c1b8e51bcc5e8691522f  60522d789a62d40d7ad1183e         CLASS   

                      userId  \
0                       Null   
1                       Null   
2                       Null   
3                       Null   
4                       Null   
5                       Null   
6                       Null   
7                       N

In [24]:
documents.authorName[documents.authorName.notnull()]

45      GARCIN
46    Hounaida
47      GARCIN
48      Maëlle
49       Gault
50      myriam
Name: authorName, dtype: object

## Homework

In [25]:
homework = pd.read_csv("data/exportHomeworkRequest-Grid view.csv")
print(homework.shape)
homework.head()

(567, 11)


id                   classId  \
0  5f4914744ded6f0001f02c9e  5f4914554ded6f0001f02c8c   
1  5f4914b64ded6f0001f02ce9  5f49139a4ded6f0001f02a08   
2  5f4915a24ded6f0001f02e3d  5f4914934ded6f0001f02cb4   
3  5f72076dfb6eb600013dda15  5f4e27ab2447780001556cf0   
4  5f665d52046ca50001ad3403  5f665d10046ca50001ad33f2   

                     userId  \
0  5f4910754ded6f0001f026a7   
1  5ec23afd7b059100010baa8b   
2  5f4911ef4ded6f0001f0271d   
3  5e5ccf59026b4800016b27b0   
4  5f665c26fb6eb600010f65eb   

                                               label              date  \
0   homework list for the day of lundi 31 août, 2020  2020-08-31 00:00   
1  Liste de devoirs pour la journée du samedi 29 ...  2020-08-29 00:00   
2  Liste de devoirs pour la journée du samedi 29 ...  2020-08-29 00:00   
3                                                NaN  2020-09-29 00:00   
4                                        Les devoirs  2020-09-22 00:00   

            type  nbRequestUsers      creationDate lastModificationDate  \
0  HOMEWORK_LIST               0  2020-05-01 02:00     2020-05-01 02:00   
1  HOMEWORK_LIST               0  2020-05-01 02:00     2020-05-01 02:00   
2  HOMEWORK_LIST               0  2020-05-01 02:00     2020-05-01 02:00   
3       HOMEWORK               0  2020-09-28 17:55     2020-09-28 17:55   
4  HOMEWORK_LIST               0  2020-09-19 21:34     2020-09-19 21:34   

                   modifier                    author  
0                      Null                      Null  
1                      Null                      Null  
2                      Null                      Null  
3  5e5ccf59026b4800016b27b0  5e5ccf59026b4800016b27b0  
4  5f665c26fb6eb600010f65eb  5f665c26fb6eb600010f65eb

In [26]:
homework.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 567 entries, 0 to 566
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   id                    567 non-null    object
 1   classId               567 non-null    object
 2   userId                567 non-null    object
 3   label                 565 non-null    object
 4   date                  567 non-null    object
 5   type                  567 non-null    object
 6   nbRequestUsers        567 non-null    int64 
 7   creationDate          567 non-null    object
 8   lastModificationDate  567 non-null    object
 9   modifier              567 non-null    object
 10  author                567 non-null    object
dtypes: int64(1), object(10)
memory usage: 48.9+ KB


In [27]:
homework.isna().sum()

id                      0
classId                 0
userId                  0
label                   2
date                    0
type                    0
nbRequestUsers          0
creationDate            0
lastModificationDate    0
modifier                0
author                  0
dtype: int64

In [28]:
homework.nbRequestUsers.unique()

array([0, 1, 2])

In [29]:
homework.where(homework!="Null")

id                   classId  \
0    5f4914744ded6f0001f02c9e  5f4914554ded6f0001f02c8c   
1    5f4914b64ded6f0001f02ce9  5f49139a4ded6f0001f02a08   
2    5f4915a24ded6f0001f02e3d  5f4914934ded6f0001f02cb4   
3    5f72076dfb6eb600013dda15  5f4e27ab2447780001556cf0   
4    5f665d52046ca50001ad3403  5f665d10046ca50001ad33f2   
..                        ...                       ...   
562  605aa0a974ec0b012da9d060  6058fe53e51bcc5e8691532b   
563  605aa0a974ec0b012da9d062  6059363be51bcc5e86915385   
564  605aa0a974ec0b012da9d064  605a2cdbe51bcc5e8691544a   
565  605aa0a974ec0b012da9d066  605a2cdbe51bcc5e8691544a   
566  605aa0a974ec0b012da9d068  605a367db787e274216edaaf   

                       userId  \
0    5f4910754ded6f0001f026a7   
1    5ec23afd7b059100010baa8b   
2    5f4911ef4ded6f0001f0271d   
3    5e5ccf59026b4800016b27b0   
4    5f665c26fb6eb600010f65eb   
..                        ...   
562  5fbfaacc79aa5d00018e8236   
563  5fbfaacc79aa5d00018e8236   
564  5fbfaacc79aa5d00018e8236   
565  5fbfaacc79aa5d00018e8236   
566  5fbfaacc79aa5d00018e8236   

                                                 label              date  \
0     homework list for the day of lundi 31 août, 2020  2020-08-31 00:00   
1    Liste de devoirs pour la journée du samedi 29 ...  2020-08-29 00:00   
2    Liste de devoirs pour la journée du samedi 29 ...  2020-08-29 00:00   
3                                                  NaN  2020-09-29 00:00   
4                                          Les devoirs  2020-09-22 00:00   
..                                                 ...               ...   
562                              I learn to use feamzy  2021-03-25 00:00   
563                                        Les devoirs  2021-03-25 00:00   
564                                        Les devoirs  2021-03-25 00:00   
565                                        Les devoirs  2021-03-25 00:00   
566                                        Les devoirs  2021-03-25 00:00   

              type  nbRequestUsers      creationDate lastModificationDate  \
0    HOMEWORK_LIST               0  2020-05-01 02:00     2020-05-01 02:00   
1    HOMEWORK_LIST               0  2020-05-01 02:00     2020-05-01 02:00   
2    HOMEWORK_LIST               0  2020-05-01 02:00     2020-05-01 02:00   
3         HOMEWORK               0  2020-09-28 17:55     2020-09-28 17:55   
4    HOMEWORK_LIST               0  2020-09-19 21:34     2020-09-19 21:34   
..             ...             ...               ...                  ...   
562       HOMEWORK               1  2021-03-24 03:15     2021-03-24 03:15   
563  HOMEWORK_LIST               1  2021-03-24 03:15     2021-03-24 03:15   
564  HOMEWORK_LIST               1  2021-03-24 03:15     2021-03-24 03:15   
565  HOMEWORK_LIST               1  2021-03-24 03:15     2021-03-24 03:15   
566  HOMEWORK_LIST               1  2021-03-24 03:15     2021-03-24 03:15   

                     modifier                    author  
0                         NaN                       NaN  
1                         NaN                       NaN  
2                         NaN                       NaN  
3    5e5ccf59026b4800016b27b0  5e5ccf59026b4800016b27b0  
4    5f665c26fb6eb600010f65eb  5f665c26fb6eb600010f65eb  
..                        ...                       ...  
562  5fbfaacc79aa5d00018e8236  5fbfaacc79aa5d00018e8236  
563  5fbfaacc79aa5d00018e8236  5fbfaacc79aa5d00018e8236  
564  5fbfaacc79aa5d00018e8236  5fbfaacc79aa5d00018e8236  
565  5fbfaacc79aa5d00018e8236  5fbfaacc79aa5d00018e8236  
566  5fbfaacc79aa5d00018e8236  5fbfaacc79aa5d00018e8236  

[567 rows x 11 columns]

## Notifications

In [69]:
notification = pd.read_csv("data/exportNotification-Grid view.csv")
print(notification.shape)
notification

(4299, 15)


id                    userId parentId  \
0     5ea70180c6cc020001891469  5e617d019fed0e00015600a9     Null   
1     5e621c689fed0e0001560199  5e6212b29fed0e00015600be     Null   
2     5ea70180c6cc020001891473  5e6212b29fed0e00015600be     Null   
3     5e839b129fed0e0001f16502  5e62341f9fed0e0001560209     Null   
4     5e6223409fed0e00015601da  5e617dbb9fed0e00015600b1     Null   
...                        ...                       ...      ...   
4294  602bdb561c2e36512e0aafdf  602bdb561c2e36512e0aafdc     Null   
4295  605934cfb787e274216ed995  605934cfb787e274216ed994     Null   
4296  605934cfb787e274216ed996  605934cfb787e274216ed994     Null   
4297  605934cfb787e274216ed997  605934cfb787e274216ed994     Null   
4298  6059363ce51bcc5e86915388  605934cfb787e274216ed994     Null   

                       eventId homeworkRequestId classId documentId  \
0     5ea70180c6cc020001891466              Null    Null       Null   
1     5e621c689fed0e0001560198              Null    Null       Null   
2     5ea70180c6cc020001891466              Null    Null       Null   
3     5e839b129fed0e0001f16500              Null    Null       Null   
4     5e6223409fed0e00015601d7              Null    Null       Null   
...                        ...               ...     ...        ...   
4294                      Null              Null    Null       Null   
4295                      Null              Null    Null       Null   
4296                      Null              Null    Null       Null   
4297                      Null              Null    Null       Null   
4298                      Null              Null    Null       Null   

                                                message  priority  \
0     Vous êtes invité à lévénement {0} organisé p...         1   
1     Vous êtes invité à L'événement c'est le week-e...         1   
2     Vous êtes invité à lévénement {0} organisé p...         1   
3     Vous êtes invité à L'événement a la maison  or...         1   
4     Vous êtes invité à L'événement Repas chez laur...         1   
...                                                 ...       ...   
4294                   Trouvez la classe de vos enfants         1   
4295                          Pour voir tous les events         1   
4296      Partagez les devoirs et l'agenda de la classe         2   
4297                   Trouvez la classe de vos enfants         1   
4298                        Hello, Welcome in class 6D.         1   

          creationDate lastModificationDate    expirationDate  \
0     2020-04-27 18:00     2021-02-15 14:07  2020-10-26 20:22   
1     2020-03-06 10:48     2021-02-15 14:07  2020-03-07 01:00   
2     2020-04-27 18:00     2021-02-15 14:07  2020-05-12 18:00   
3     2020-03-31 21:33     2021-02-15 14:07  2020-04-10 02:00   
4     2020-03-06 11:17     2021-02-15 14:07  2020-03-20 11:17   
...                ...                  ...               ...   
4294  2021-03-23 20:05     2021-03-23 20:05  2021-03-03 15:48   
4295  2021-03-23 01:22     2021-03-23 01:22  2021-04-07 01:22   
4296  2021-03-23 01:22     2021-03-23 01:25  2021-03-23 01:25   
4297  2021-03-23 01:22     2021-03-23 01:26  2021-03-23 01:26   
4298  2021-03-23 01:28     2021-03-23 01:28  2021-04-07 01:28   

            notificationType  sphere                             title  
0           INVITATION_EVENT   EVENT                               NaN  
1           INVITATION_EVENT     NaN                               NaN  
2           INVITATION_EVENT   EVENT                               NaN  
3           INVITATION_EVENT     NaN                               NaN  
4           INVITATION_EVENT     NaN                               NaN  
...                      ...     ...                               ...  
4294             SETUP_CLASS  SCHOOL  Trouvez la classe de vos enfants  
4295          SETUP_CALENDAR    HOME          Synchronisez vos agendas  
4296            SETUP_SCHOOL  SCHOOL         Découvrez la sphère école  
4297  

In [31]:
notification.sphere.unique()

array(['EVENT', nan, 'HOME', 'SCHOOL'], dtype=object)

In [32]:
notification.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4299 entries, 0 to 4298
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   id                    4299 non-null   object
 1   userId                4299 non-null   object
 2   parentId              4299 non-null   object
 3   eventId               4299 non-null   object
 4   homeworkRequestId     4299 non-null   object
 5   classId               4299 non-null   object
 6   documentId            4299 non-null   object
 7   message               4289 non-null   object
 8   priority              4299 non-null   int64 
 9   creationDate          4299 non-null   object
 10  lastModificationDate  4299 non-null   object
 11  expirationDate        4299 non-null   object
 12  notificationType      4299 non-null   object
 13  sphere                4290 non-null   object
 14  title                 3130 non-null   object
dtypes: int64(1), object(14)
memory usage: 

In [33]:
notification[~(notification["classId"]=="Null")]

id                    userId parentId eventId  \
472   5f68ba8afb6eb600010f68c1  5e5e2cb73dc85e0001e90672     Null    Null   
473   5f68c1a0046ca50001ad371c  5e5ccf59026b4800016b27b0     Null    Null   
474   5f68ba8afb6eb600010f68c4  5f1099c1287d3e0001cc75b9     Null    Null   
475   5f68ba8afb6eb600010f68c2  5f64de18722f7500016ed76b     Null    Null   
478   5f68c1da046ca50001ad372c  5e5ccf59026b4800016b27b0     Null    Null   
...                        ...                       ...      ...     ...   
4238  605458619a62d40d7ad1228f  60511b429a62d40d7ad11624     Null    Null   
4239  605458619a62d40d7ad12291  60511b429a62d40d7ad11624     Null    Null   
4252  6058be4be51bcc5e869151f4  60522fe49a62d40d7ad118b0     Null    Null   
4253  6058be4be51bcc5e869151f6  605236719a62d40d7ad11977     Null    Null   
4255  6058c1b8e51bcc5e86915230  605226efa0bad44c9a6c9b54     Null    Null   

             homeworkRequestId                   classId  \
472   5f68ba8afb6eb600010f68c0  5f4e27ab2447780001556cf0   
473   5f68ba8afb6eb600010f68c0  5f4e27ab2447780001556cf0   
474   5f68ba8afb6eb600010f68c0  5f4e27ab2447780001556cf0   
475   5f68ba8afb6eb600010f68c0  5f4e27ab2447780001556cf0   
478   5f68ba8afb6eb600010f68c0  5f4e27ab2447780001556cf0   
...                        ...                       ...   
4238  605458619a62d40d7ad1228e  60511ddc9a62d40d7ad11638   
4239  605458619a62d40d7ad12290  60511ddc9a62d40d7ad11638   
4252  6058be4be51bcc5e869151f3  60522d789a62d40d7ad1183e   
4253  6058be4be51bcc5e869151f3  60522d789a62d40d7ad1183e   
4255  6058be4be51bcc5e869151f3  60522d789a62d40d7ad1183e   

                    documentId  \
472                       Null   
473   5f68c1a0046ca50001ad371b   
474                       Null   
475                       Null   
478   5f68c1d9046ca50001ad372b   
...                        ...   
4238                      Null   
4239                      Null   
4252                      Null   
4253                      Null   
4255  6058c1b8e51bcc5e8691522f   

                                                message  priority  \
472                              Type de devoir inconnu         1   
473   Marie Lewin a déposé l'exercice de Maths exerc...         1   
474                              Type de devoir inconnu         1   
475                              Type de devoir inconnu         1   
478   Marie Lewin a déposé l'exercice de Maths exerc...         1   
...                                                 ...       ...   
4238  le coach de feamzy, a besoin des devoirs de la...         1   
4239  le coach de feamzy, a besoin de l'exercice 'I ...         1   
4252  Nathalie Lassère, a besoin des devoirs de la c...         1   
4253  Nathalie Lassère, a besoin des devoirs de la c...         1   
4255  myriam bonierbale a déposé la liste des devoir...         1   

          creationDate lastModificationDate    expirationDate  \
472   2020-09-21 16:36     2021-02-15 14:07  2020-09-21 17:08   
473   2020-09-21 17:07     2021-02-15 14:07  2020-09-22 02:00   
474   2020-09-21 16:36     2021-02-15 14:07  2021-01-06 16:14   
475   2020-09-21 16:36     2021-02-15 14:07  2020-09-21 17:08   
478   2020-09-21 17:08     2021-02-15 14:07  2020-09-22 02:00   
...                ...                  ...               ...   
4238  2021-03-19 08:53     2021-03-19 08:53  2021-03-18 23:59   
4239  2021-03-19 08:53     2021-03-19 08:53  2021-03-19 23:59   
4252  2021-03-22 16:56     2021-03-22 17:11  2021-03-22 17:11   
4253  2021-03-22 16:56     2021-03-22 17:11  2021-03-22 17:11   
4255  2021-03-22 17:11     2021-03-22 17:11  2021-03-23 23:59   

       notificationType  sphere title  
472    HOMEWORK_REQUEST  SCHOOL   NaN  
473   HOMEWORK_RESPONSE  SCHOOL   NaN  
474    HOMEWORK_REQUEST  SCHOOL   NaN  
475    HOMEWORK_REQUEST  SCHOOL   NaN  
478   HOMEWORK_RESPONSE  SCHOOL   NaN  
...                 ...     ...   ...  
4238   HOMEWORK_REQUEST  SCHOOL   NaN  
4239   HOMEWORK_REQUEST  SCHOOL   NaN  
425

In [70]:
notification.message.unique()

array(['Vous êtes invité Ã\xa0 lévénement {0} organisé par {1}',
       "Vous êtes invité à L'événement c'est le week-end ! organisé par [Samuel Rémy]",
       "Vous êtes invité à L'événement a la maison  organisé par [Jérémie Lachcar]",
       "Vous êtes invité à L'événement Repas chez laura  organisé par [Laura Michael]",
       "Vous êtes invité à L'événement 1 er test caro organisé par [Caroline Guérard]",
       "Vous êtes invité à L'événement Les lyonnais à Paris organisé par [Marie Lewin]",
       'You are invited to the event Réunion feamzy organized by [Marie Lewin]',
       'You are invited to the event break en normandie organized by [Simon Guerard]',
       "L'événement a été modifié",
       "Vous êtes invité à L'événement Un petit télé apéro ?  organisé par [Marie Lewin]",
       "Vous êtes invité à L'événement hello test organisé par [Jérémie Lachcar]",
       'You are invited to the event Côte de boeuf party organized by [Marie Lewin]',
       nan,
       'You are invi

In [68]:
notification.message = notification.message.str.replace("ê","ê")
notification.message = notification.message.str.replace("Ã","à")
notification.message = notification.message.replace("lévénement","l'événement")
notification.message = notification.message.replace("L'événement","l'événement")
notification.message = notification.message.replace("  lévénement ","à l'événement")
notification.message

0       Vous êtes invité à  lévénement {0} organisé pa...
1       Vous êtes invité à L'événement c'est le week-e...
2       Vous êtes invité à  lévénement {0} organisé pa...
3       Vous êtes invité à L'événement a la maison  or...
4       Vous êtes invité à L'événement Repas chez laur...
                              ...                        
4294                     Trouvez la classe de vos enfants
4295                            Pour voir tous les events
4296        Partagez les devoirs et l'agenda de la classe
4297                     Trouvez la classe de vos enfants
4298                          Hello, Welcome in class 6D.
Name: message, Length: 4299, dtype: object

In [67]:
print(notification.message[0])

Vous êtes invité à  lévénement {0} organisé par {1}


In [50]:
notification.message.str.encode('latin1')

0       b'Vous \xc3\xaates invit\xe9 \xc3\xa0 l\xe9v\x...
1       b"Vous \xeates invit\xe9 \xe0 L'\xe9v\xe9nemen...
2       b'Vous \xc3\xaates invit\xe9 \xc3\xa0 l\xe9v\x...
3       b"Vous \xeates invit\xe9 \xe0 L'\xe9v\xe9nemen...
4       b"Vous \xeates invit\xe9 \xe0 L'\xe9v\xe9nemen...
                              ...                        
4294                  b'Trouvez la classe de vos enfants'
4295                         b'Pour voir tous les events'
4296     b"Partagez les devoirs et l'agenda de la classe"
4297                  b'Trouvez la classe de vos enfants'
4298                       b'Hello, Welcome in class 6D.'
Name: message, Length: 4299, dtype: object

## Conclusions

**Tables structures**
- Users: 2928 rows x 12 columns
- Classes: 313 rows x 28 columns
- Homeworks: 567 rows x 11 columns
- Events: 2426 rows x 22 columns
- Notifications: 4299 rows x 15 columns
- Documents: 51 rows x 13 columns


**Null values:** Lot's of columns with null values. Some are NaN identified by python and in the `Notifications` table, the values are indicated as a tring "Null".
- To be careful with: 